### PR TITLE
Add support for cypress-based tests

### PR DESCRIPTION
**Pull request by Fewwy**
Implements https://issues.redhat.com/browse/RHIF-71.

This adds support for the cypress component and e2e tests (version 10). Also utilizes functions and constants from `@redhat-cloud-services/frontend-components-utils`.

## How to test

Run `npm run test` (will run together with jest tests) or `npm run test:ct` (cypress tests alone). You should obtain the results of RulesTable.cy.js test.

### CODE REVIEW

File: cypress/support/commands.js
Comment (gkarat) [inline cypress/support/commands.js:26]: Hm, can we try to put this `ouiaId` and `ouiaType` to the common utils?

File: src/PresentationalComponents/RulesTable/RulesTable.spec.ct.js
Comment (gkarat) [inline src/PresentationalComponents/RulesTable/RulesTable.spec.ct.js:6]:
```suggestion
import { IntlProvider } from '@redhat-cloud-services/frontend-components-translations';
```
because the trailing slash can sometimes lead to problems with package discovery

File: src/PresentationalComponents/RulesTable/RulesTable.spec.ct.js
Comment (gkarat) [inline src/PresentationalComponents/RulesTable/RulesTable.spec.ct.js:46]: perhaps we could use this https://github.com/RedHatInsights/insights-advisor-frontend/blob/master/src/AppConstants.js#L11?

File: src/PresentationalComponents/RulesTable/RulesTable.spec.ct.js
Comment (gkarat) [inline src/PresentationalComponents/RulesTable/RulesTable.spec.ct.js:55]: this could be useful in other tests => move to common utils?

File: src/PresentationalComponents/RulesTable/RulesTable.spec.ct.js
Comment (gkarat) [inline src/PresentationalComponents/RulesTable/RulesTable.spec.ct.js:70]: please also double check that ocp-advisor rule categories are identical to what in the rhel ones

File: src/PresentationalComponents/RulesTable/RulesTable.spec.ct.js
Comment (gkarat) [inline src/PresentationalComponents/RulesTable/RulesTable.spec.ct.js:169]: 👍🏼

File: src/PresentationalComponents/RulesTable/RulesTable.spec.ct.js
Comment (gkarat) [inline src/PresentationalComponents/RulesTable/RulesTable.spec.ct.js:173]: `transforms` seems obsolete for the tests. why we have to use it?

File: src/PresentationalComponents/RulesTable/RulesTable.spec.ct.js
Comment (gkarat) [inline src/PresentationalComponents/RulesTable/RulesTable.spec.ct.js:172]: while declaring test data, can we avoid using intl? I mean, in tests, the language should always be English, and these setups look unnecessary

File: src/PresentationalComponents/RulesTable/RulesTable.spec.ct.js
Comment (gkarat) [inline src/PresentationalComponents/RulesTable/RulesTable.spec.ct.js:201]: I remember this was the default value in test utils. Do we need to duplicate it here?

File: src/PresentationalComponents/RulesTable/RulesTable.spec.ct.js
Comment (gkarat) [inline src/PresentationalComponents/RulesTable/RulesTable.spec.ct.js:642]: these two passed to me => uncomment?

File: config/cypress.webpack.config.js
Comment (Fewwy) [reply config/cypress.webpack.config.js:7]: You are correct, I removed it

File: cypress/support/commands.js
Comment (Fewwy) [reply cypress/support/commands.js:26]: Yes, will add them there with the next update

File: src/PresentationalComponents/RulesTable/RulesTable.spec.ct.js
Comment (Fewwy) [reply src/PresentationalComponents/RulesTable/RulesTable.spec.ct.js:6]: You are correct. removed it

File: src/PresentationalComponents/RulesTable/RulesTable.spec.ct.js
Comment (Fewwy) [reply src/PresentationalComponents/RulesTable/RulesTable.spec.ct.js:46]: Don't really understand what you meant by that.

File: src/PresentationalComponents/RulesTable/RulesTable.spec.ct.js
Comment (Fewwy) [reply src/PresentationalComponents/RulesTable/RulesTable.spec.ct.js:70]: We can add that to the utility library with the next update and import them from there. But i'm not sure that it's the best solution :)

File: src/PresentationalComponents/RulesTable/RulesTable.spec.ct.js
Comment (Fewwy) [reply src/PresentationalComponents/RulesTable/RulesTable.spec.ct.js:173]: You are correct, removed it

File: src/PresentationalComponents/RulesTable/RulesTable.spec.ct.js
Comment (Fewwy) [reply src/PresentationalComponents/RulesTable/RulesTable.spec.ct.js:172]: Do you suggest to replace them by strings with column names?

File: src/PresentationalComponents/RulesTable/RulesTable.spec.ct.js
Comment (Fewwy) [reply src/PresentationalComponents/RulesTable/RulesTable.spec.ct.js:201]: You are correct, that's unnecessary

File: src/PresentationalComponents/RulesTable/RulesTable.spec.ct.js
Comment (Fewwy) [reply src/PresentationalComponents/RulesTable/RulesTable.spec.ct.js:642]: Are you sure? 
image

File: src/PresentationalComponents/RulesTable/RulesTable.spec.ct.js
Comment (gkarat) [reply src/PresentationalComponents/RulesTable/RulesTable.spec.ct.js:46]: @Fewwy , I mean this looks like a duplicate code and, in the file I sent, there is already an `intl` instantiated. We could just make an export of it and then import in your tests

File: src/PresentationalComponents/RulesTable/RulesTable.spec.ct.js
Comment (gkarat) [reply src/PresentationalComponents/RulesTable/RulesTable.spec.ct.js:172]: yes, replacing with what is `formatMessage` evaluated with default setup (English)

File: src/PresentationalComponents/RulesTable/RulesTable.spec.ct.js
Comment (gkarat) [reply src/PresentationalComponents/RulesTable/RulesTable.spec.ct.js:642]: yes, it did worked for me. but whatever, let's check it later and not be blocked by this now

File: cypress/support/commands.js
Comment (adonispuente) [inline cypress/support/commands.js:1]: Do we want to keep all this commented out example stuff?

File: cypress/support/commands.ts
Comment (adonispuente) [inline cypress/support/commands.ts:1]: Is this file necessary?

File: src/PresentationalComponents/Loading/Loading.spec.ct.js
Comment (adonispuente) [inline src/PresentationalComponents/Loading/Loading.spec.ct.js:1]: When I open cypress, I dont see this test available. Only the rulesTable. Is this intentional because of how basic the test is? If so we should just remove it

File: src/PresentationalComponents/RulesTable/RulesTable.cy.js
Comment (adonispuente) [inline src/PresentationalComponents/RulesTable/RulesTable.cy.js:1]: In the future I would like to work with you on compartmentalizing as much as we can. This is the first implementation so this is fine for now, after this is merged, in the next refactor PR I'd like to work on this.

**Comment by adonispuente** [comment]
Travis is mad about some test failing, but this has progressed very well!

File: src/PresentationalComponents/Loading/Loading.spec.ct.js
Comment (Fewwy) [reply src/PresentationalComponents/Loading/Loading.spec.ct.js:1]: Forgot to change "spec.ct.js" to "cy.js"

File: cypress/support/commands.ts
Comment (Fewwy) [reply cypress/support/commands.ts]: This file was added in migration to v10, I removed it

File: cypress/support/commands.js
Comment (Fewwy) [reply cypress/support/commands.js:1]: We can remove it, but it's doesn't hurt anyone :)

**Review by adonispuente — DISMISSED** [review]
I would still recommend removing commented out examples just to clean things up, but the PR is working properly and as intended, great job! LGTM!

**Review by gkarat — CHANGES_REQUESTED** [review]
Unfortunately, the jenkins build is broken. See "Your system is missing the dependency: Xvfb": https://ci.int.devshift.net/job/RedHatInsights-insights-advisor-frontend-pr-check/485/console.

**Review by gkarat — APPROVED** [review]

**Pull request merged**
